modules/reactor/src/meshgenerators/TriPinHexAssemblyGenerator.C

```diff
@@ -0,0 +1,507 @@
+  {
+    meshes.push_back(buildSinglePinSection(_side_length,
+                                           _ring_offset,
+                                           _ring_radii[i],
+                                           _ring_intervals[i],
+                                           _has_rings[i],
+                                           _preserve_volumes,
+                                           _num_sectors_per_side,
+                                           _background_intervals,
+                                           block_ids_new[i],
+                                           _node_id_background_meta));
+    if (!_pin_id_name.empty())
+      meshes[i]->add_elem_integer(_pin_id_name, true, _pin_id_values[i]);
+    if (i > 0)
+    {
+      MeshTools::Modification::rotate(*meshes[i], 120.0 * (Real)i, 0, 0);
+      meshes[0]->stitch_meshes(*std::move(meshes[i]),
+                               OUTER_SIDESET_ID,
+                               OUTER_SIDESET_ID,
+                               TOLERANCE,
+                               /*clear_stitched_boundary_ids=*/true);
+    }
+  }
+
+  if (_assembly_orientation == AssmOrient::pin_up)
+    MeshTools::Modification::rotate(*meshes[0], 90, 0, 0);
+  else
+    MeshTools::Modification::rotate(*meshes[0], 270, 0, 0);
+  /* Add subdomain names */
+  for (const auto & block_info_pair : block_info)
+    meshes[0]->subdomain_name(block_info_pair.first) = block_info_pair.second;
+  if (_external_boundary_id > 0)
+    MooseMesh::changeBoundaryId(*meshes[0], OUTER_SIDESET_ID, _external_boundary_id, false);
+  if (!_external_boundary_name.empty())
+  {
+    meshes[0]->get_boundary_info().sideset_name(
+        _external_boundary_id > 0 ? _external_boundary_id : (boundary_id_type)OUTER_SIDESET_ID) =
+        _external_boundary_name;
+    meshes[0]->get_boundary_info().nodeset_name(
+        _external_boundary_id > 0 ? _external_boundary_id : (boundary_id_type)OUTER_SIDESET_ID) =
+        _external_boundary_name;
+  }
+  return dynamic_pointer_cast<MeshBase>(meshes[0]);
+}
+
+std::unique_ptr<ReplicatedMesh>
+TriPinHexAssemblyGenerator::buildSinglePinSection(
+    const Real side_length,
+    const Real ring_offset,
+    const std::vector<Real> ring_radii,
+    const std::vector<unsigned int> ring_intervals,
+    const bool has_rings,
+    const bool preserve_volumes,
+    const unsigned int num_sectors_per_side,
+    const unsigned int background_intervals,
+    const std::vector<subdomain_id_type> block_ids_new,
+    dof_id_type & node_id_background_meta)
+{
+  const Real secondary_side_length_0(side_length / 2.0 - ring_offset);
+  const Real secondary_side_length_1(
+      std::sqrt(side_length * side_length / 4.0 * 3.0 + ring_offset * ring_offset));
+  const Real primary_side_length_0(
+      std::sqrt(side_length * side_length / 4.0 * 3.0 + ring_offset * ring_offset));
+  const Real primary_side_length_1(side_length / 2.0 + ring_offset);
+  const Real azimuthal_angle_0(
```

Comment:
add comments so we can see immediately what
- azimuthal_angle
- rotation_angle
- alpha_angle
- azimuthal_tangeant
- azimuthal_list
are

Reply:
`azimuthal_tangeant` is actually not needed. Will update the code to include comments about other variables.